src/Analyzer.Utilities/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -144,7 +154,12 @@ public static Accessibility DetermineMinimalAccessibility(this ITypeSymbol typeS
         {
             return typeSymbol.Accept(MinimalAccessibilityVisitor.Instance);
         }
-
+        
+        public static bool IsTypeDeclaredInExpectedAssembly(ITypeSymbol typeSymbol, string assemblyName)
```

Comment:
Please move this method to Desktop analyzer's helper class, it is not likely to be used by other analyzers.